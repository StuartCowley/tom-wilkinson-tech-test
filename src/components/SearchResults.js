import React from "react";
import "../styles/searchResults.css";

const SearchResults = ({results}
) => {

  if(!results.length) {
    return <p> No results <br></br><br></br>
    Please try another search </p> 
  } else

  return (
    <>
    <div className="image-gallery">
      {results.map((image) => (
        <img className="return-images" src={image} alt="spaceImage" key={image} />
      ))}
      </div>
    </>

  );
}

export default SearchResults;
