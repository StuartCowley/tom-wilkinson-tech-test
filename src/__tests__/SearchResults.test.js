import React from "react";
import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("Search Results", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<SearchResults results={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders with results", () => {
    const results = ["search-image1", "search-image2", "search-image3"];
    const { asFragment } = render(<SearchResults results={results} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders no results message when there are no results", () => {
    render(<SearchResults results={[]} hasSearched={true} />);
    const noResultsMessage = screen.getByText(
      "No results" && "Please try another search"
    );
    expect(noResultsMessage).toBeInTheDocument();
  });
  it("renders images when there are results to return", () => {
    const results = [
      "search-image1",
      "search-image2",
      "search-image3",
      "search-image4]",
    ];
    const hasSearched = true;
    render(<SearchResults results={results} hasSearched={hasSearched} />);
    const imageResults = screen.queryAllByAltText("spaceImage");
    expect(imageResults.length).toBe(4);
  });
});

//2 snapshots - one with results as [], one with array as strings
//set up as sperate tests, next.js
