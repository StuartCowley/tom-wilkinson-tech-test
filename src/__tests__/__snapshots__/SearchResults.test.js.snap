// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results renders correctly 1`] = `<DocumentFragment />`;

exports[`Search Results renders with results 1`] = `<DocumentFragment />`;
